drivers/SmartThings/harman-luxury/src/handlers.lua

```diff
@@ -0,0 +1,245 @@
+local capabilities = require "st.capabilities"
+local api = require "api.apis"
+local log = require "log"
+
+local Handler = {}
+
+--- handler of switch.on
+---@param driver Driver
+---@param device Device
+---@param cmd function|nil
+function Handler.handle_on(driver, device, cmd)
+    log.info("Starting handle_on")
+    -- send API switch on message
+    local ip = device:get_field("device_ipv4")
+    local ret = api.SetOn(ip)
+    if type(ret) == "table" then
+        log.warn("Error during handle_on(): " .. ret["error"])
+    end
+    -- verify change and update app
+    local ret = api.GetPowerState(ip)
+    if ret == "online" then
+        device:emit_event(capabilities.switch.switch.on())
+    else
+        device:emit_event(capabilities.switch.switch.off())
+    end
+end
+
+--- handler of switch.off
+---@param driver Driver
+---@param device Device
+---@param cmd function|nil
+function Handler.handle_off(driver, device, cmd)
+    log.info("Starting handle_off")
+    -- send API switch off message
+    local ip = device:get_field("device_ipv4")
+    local ret = api.SetOff(ip)
+    if type(ret) == "table" then
+        log.warn("Error during handle_off(): " .. ret["error"])
+    end
+    -- verify change and update app
+    local ret = api.GetPowerState(ip)
+    if ret == "online" then
+        device:emit_event(capabilities.switch.switch.on())
+    else
+        device:emit_event(capabilities.switch.switch.off())
+    end
+end
+
+function Handler.handle_mute(driver, device, cmd)
+    log.info("Starting handle_mute")
+    -- send API mute on message
+    local ip = device:get_field("device_ipv4")
+    local ret = api.Mute(ip)
+    if type(ret) == "table" then
+        log.warn("Error during handle_mute(): " .. ret["error"])
+    end
+    -- verify change and update app
+    local ret = api.GetMute(ip)
+    if ret then
+        device:emit_event(capabilities.audioMute.mute.muted())
+    else
+        device:emit_event(capabilities.audioMute.mute.unmuted())
+    end
+end
+
+function Handler.handle_unmute(driver, device, cmd)
+    log.info("Starting handle_unmute")
+    -- send API mute off message
+    local ip = device:get_field("device_ipv4")
+    local ret = api.Unmute(ip)
+    if type(ret) == "table" then
+        log.warn("Error during handle_unmute(): " .. ret["error"])
+    end
+    -- verify change and update app
+    local ret = api.GetMute(ip)
+    if ret then
+        device:emit_event(capabilities.audioMute.mute.muted())
+    else
+        device:emit_event(capabilities.audioMute.mute.unmuted())
+    end
+end
+
+function Handler.handle_set_mute(driver, device, cmd)
```

Comment:
again, since you've already defined mute and unmute methods, this method seems reduplicative